<a href="https://colab.research.google.com/github/redinbluesky/fun-coding-python-data-analysis/blob/main/08-실습:부동산_데이터_분석하기.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#  목차
* [Chapter 8-1 데이터 확인하기](#chapter8-1)
    * [Chapter 8-1-1 데이터 구조 파악하기](#chapter8-1-1)
    * [Chapter 8-1-2 기본 통계 정보 확인하기](#chapter8-1-2)
* [Chapter 8-2 부동산 매매가격 분석하기](#chapter8-2)    
    * [Chapter 8-2-1 평균 매매가격 계산하기](#chapter8-2-1)    
* [Chapter 8-3 셀프 체크](#chapter8-3)        

## Chapter 8-1 데이터 확인하기 <a class="anchor" id="chapter8-1"></a>
1. 서울 열린데이터광장에서 제공하는 서울특별시 부동산 실거래 정보를 기반으로 한 데이터를 사용한다.
    - 파일명: seoul_apartment.csv

2. 데이터셋을 얻을 수 있는 사이트
    - 공공데이터 포털
        - https://www.data.go.kr/
        - 대한민국 정부에서 운영하는 공공데이터 포털로, 다양한 분야의 데이터를 제공한다.
    - 서울 열린데이터광장
        - https://data.seoul.go.kr/
        - 서울시에서 제공하는 다양한 데이터를 제공하는 플랫폼으로, 행정, 인구, 교통, 환경 등 다양한 분야의 데이터를 제공한다.
    - Kaggle
        - https://www.kaggle.com/
        - 데이터 과학자와 머신러닝 엔지니어들이 데이터를 공유하고 분석하는 플랫폼으로, 다양한 분야의 데이터셋을 제공한다.
    - 구글 데이터셋 검색
        - https://datasetsearch.research.google.com/
        - 구글에서 제공하는 데이터셋 검색 엔진으로, 다양한 분야의 데이터셋을 검색할 수 있다.
    - UCI 머신러닝 저장소
        - https://archive.ics.uci.edu/ml/index.php
        - 머신러닝 연구에 자주 사용되는 데이터셋을 제공하는 저장소로, 다양한 분야의 데이터셋을 제공한다.
    - 세계은행 데이터
        - https://data.worldbank.org/
        - 세계은행에서 제공하는 다양한 경제 및 사회 데이터를 제공하는 플랫폼으로, 다양한 분야의 데이터셋을 제공한다.
    - UN 데이터
        - https://data.un.org/
        - 유엔에서 제공하는 다양한 국제 데이터를 제공하는 플랫폼으로, 다양한 분야의 데이터셋을 제공한다.

### Chapter 8-1-1 데이터 구조 파악하기 <a class="anchor" id="chapter8-1-1"></a>

In [1]:
# Pandas로 seoul_apartment.csv 파일 불러오기
import pandas as pd

# CSV 파일 경로
csv_file_path = "./data/seoul_apartment.csv"

# Panndas로 csv 파일 읽어오기
df = pd.read_csv(csv_file_path)

# head() 메서드로 처음 5개행 데이터 보기
df.head()

,NO,시군구,번지,본번,부번,단지명,전용면적(㎡),계약년월,계약일,거래금액(만원),층,건축년도,도로명
0,1,서울특별시 종로구 사직동,9,9,0,광화문스페이스본(101동~105동),94.51,200811,3,63000,6,2008,사직로8길 4
1,2,서울특별시 종로구 사직동,9,9,0,광화문스페이스본(101동~105동),146.92,200810,10,97000,13,2008,사직로8길 4
2,3,서울특별시 종로구 사직동,9,9,0,광화문스페이스본(101동~105동),146.92,200810,7,100000,1,2008,사직로8길 4
3,4,서울특별시 종로구 사직동,9,9,0,광화문스페이스본(101동~105동),146.92,200810,1,95000,12,2008,사직로8길 4
4,5,서울특별시 종로구 사직동,9,9,0,광화문스페이스본(101동~105동),146.92,200809,26,95000,11,2008,사직로8길 4


In [2]:
# tail() 메서드로 마지막 5개행 데이터 보기
df.tail()

,NO,시군구,번지,본번,부번,단지명,전용면적(㎡),계약년월,계약일,거래금액(만원),층,건축년도,도로명
1260221,54409,서울특별시 강동구 강일동,670,670,0,강일리버파크5단지,84.83,201101,10,34907,1,2009,아리수로97길 20
1260222,54410,서울특별시 강동구 강일동,670,670,0,강일리버파크5단지,84.83,201101,8,42000,4,2009,아리수로97길 20
1260223,54411,서울특별시 강동구 강일동,673,673,0,강일리버파크4단지,84.74,201101,6,36889,2,2009,아리수로97길 19
1260224,54412,서울특별시 강동구 강일동,670,670,0,강일리버파크5단지,84.74,201101,5,42000,8,2009,아리수로97길 20
1260225,54413,서울특별시 강동구 강일동,670,670,0,강일리버파크5단지,59.87,201101,3,19006,7,2009,아리수로97길 20


In [4]:
# 데이터 크기 확인
df.shape

(1260226, 13)

### Chapter 8-1-2 기본 통계정보 확인하기 <a class="anchor" id="chapter8-1-2"></a>

In [5]:
# describe() 메서드를 통해 기본정보 확인하기
df.describe()

,NO,본번,부번,전용면적(㎡),계약년월,계약일,거래금액(만원),층,건축년도
count,1.260226e+06,1.260226e+06,1.260226e+06,1.260226e+06,1.260226e+06,1.260226e+06,1.260226e+06,1.260226e+06,1.260226e+06
mean,4.040072e+04,5.686869e+02,5.924721e+00,7.663131e+01,2.014313e+05,1.582225e+01,5.692392e+04,8.859588e+00,1.998733e+03
std,2.830956e+04,5.187457e+02,4.618219e+01,2.914868e+01,4.842430e+02,8.733074e+00,4.763579e+04,5.967147e+00,9.282777e+00
min,1.000000e+00,1.000000e+00,0.000000e+00,1.002000e+01,2.006010e+05,1.000000e+00,3.500000e+02,-4.000000e+00,1.961000e+03
25%,1.696900e+04,1.890000e+02,0.000000e+00,5.959000e+01,2.010100e+05,8.000000e+00,2.900000e+04,4.000000e+00,1.992000e+03
50%,3.550600e+04,4.760000e+02,0.000000e+00,8.037000e+01,2.015050e+05,1.600000e+01,4.300000e+04,8.000000e+00,2.000000e+03
75%,5.987875e+04,7.830000e+02,1.000000e+00,8.496000e+01,2.018030e+05,2.300000e+01,6.880000e+04,1.200000e+01,2.005000e+03
max,1.198730e+05,4.974000e+03,2.837000e+03,4.243200e+02,2.024050e+05,3.100000e+01,1.800000e+06,6.900000e+01,2.023000e+03


In [6]:
# set_option() 함수를 사용해 출력 형식변경
pd.set_option('display.float_format', '{:,.0f}'.format)

In [7]:
# describe() 메서드를 통해 기본정보 확인하기
df.describe()

,NO,본번,부번,전용면적(㎡),계약년월,계약일,거래금액(만원),층,건축년도
count,"1,260,226","1,260,226","1,260,226","1,260,226","1,260,226","1,260,226","1,260,226","1,260,226","1,260,226"
mean,"40,401",569,6,77,"201,431",16,"56,924",9,"1,999"
std,"28,310",519,46,29,484,9,"47,636",6,9
min,1,1,0,10,"200,601",1,350,-4,"1,961"
25%,"16,969",189,0,60,"201,010",8,"29,000",4,"1,992"
50%,"35,506",476,0,80,"201,505",16,"43,000",8,"2,000"
75%,"59,879",783,1,85,"201,803",23,"68,800",12,"2,005"
max,"119,873","4,974","2,837",424,"202,405",31,"1,800,000",69,"2,023"


1. 요약정보 확인
    - 모든 열의 카운트가 같은걸로 보아 모든 열에서 결측치가 없다.
    - 전용면적의 평균은 77 제곱미터이며, 평균 거래금액은 약 5억 6천만원이다.
    - 거래금액은 최소 350만원에서 180억까지 다양하게 분포되어 있다.

2. set_option() 함수
    - Pandas에서 데이터 출력 형식을 설정하거나 제어하는데 사용한다.
    - display.max_rows: 출력할 최대 행 수를 설정한다. 기본값은 60이이다.
    - display.max_columns: 출력할 최대 열 수를 설정한다. 기본값은 20이다.
    - display.max_colwidth: 열의 최대 너비를 설정한다. 기본값은 50이며, 이보다 긴 문자열은 '...'로 표시된다.
    - display.precision: 숫자의 소수점 이하 자릿수를 설정한다. 기본값은 6이다.
    - display.float_format: 부동소수점 숫자의 출력 형식을 설정한다.

## Chapter 8-2 부동산 매매가격 분석하기 <a class="anchor" id="chapter8-2"></a>
### Chapter 8-2-1 평균 매매가격 계산하기 <a class="anchor" id="chapter8-2-1"></a>

In [ ]:
# 부동산 매매 가격의 평균 계산하기
average_price = df['거래금액(만원)'].mean()
print("평균 매매 가격: ", average_price)

평균 매매 가격:  56923.91560402658


In [11]:
# 각 시군구의 평균 매매가격 계산하기
grouped = df.groupby('시군구')
average_price = grouped['거래금액(만원)'].mean()
print("지역별 평균 매매가격:")
average_price

지역별 평균 매매가격:


시군구
서울특별시 강남구 개포동    92,325
서울특별시 강남구 논현동    79,531
서울특별시 강남구 대치동   139,215
서울특별시 강남구 도곡동   135,587
서울특별시 강남구 삼성동   117,267
                  ...  
서울특별시 중랑구 면목동    33,889
서울특별시 중랑구 묵동     37,848
서울특별시 중랑구 상봉동    39,156
서울특별시 중랑구 신내동    31,648
서울특별시 중랑구 중화동    33,364
Name: 거래금액(만원), Length: 338, dtype: float64

In [12]:
# 각 단지의 최대 전용면적을 구한다.
max_ares = df.groupby('단지명')['전용면적(㎡)'].max()
print("단지별 최대 전용면적: ")
max_ares

단지별 최대 전용면적: 


단지명
(1-10)          76
(1-102)         70
(1101-1)        15
(143-37)        18
(1544-0)        82
                ..
힐튼빌리지1차        143
힐튼빌리지2차        157
힐하우스           242
힐하우스(577-79)   101
힐하우스(802-3)    222
Name: 전용면적(㎡), Length: 7797, dtype: float64

In [13]:
# 건축년도별 거래 횟수 구하기
yearly_transactions = df.groupby('건축년도')['NO'].count()
print("건축년도별 거래 횟수: ")
print(yearly_transactions)

건축년도별 거래 횟수: 
건축년도
1961       67
1965       17
1966       66
1967       22
1968      355
1969      600
1970      739
1971     2607
1972      678
1973      770
1974     3237
1975     2642
1976     2815
1977     3061
1978     8600
1979     6497
1980    10182
1981     5985
1982    12158
1983    17305
1984    12197
1985     9559
1986    23361
1987    26070
1988    66899
1989    28877
1990    17419
1991    18396
1992    37383
1993    35326
1994    28005
1995    32845
1996    36839
1997    47654
1998    52369
1999    69498
2000    65827
2001    49057
2002    48505
2003    74306
2004    75645
2005    53575
2006    38316
2007    30414
2008    38739
2009    24841
2010    21569
2011    21925
2012    17875
2013    14803
2014    18693
2015     9735
2016     9323
2017     5347
2018     5669
2019     6227
2020     4170
2021     2735
2022     1496
2023      334
Name: NO, dtype: int64


## Chapter 8-3 셀프체크 <a class="anchor" id="chapter8-3"></a>

In [14]:
# 거래금액(만원) 열의 최솟값, 최댓값, 평균, 중앙값을 계산해 출력하는 코드를 작성
df['거래금액(만원)'].min(), df['거래금액(만원)'].max(), df['거래금액(만원)'].mean(), df['거래금액(만원)'].median()

(np.int64(350),
 np.int64(1800000),
 np.float64(56923.91560402658),
 np.float64(43000.0))

In [21]:
# 시군구 열을 기준으로 각 지역의 평균 거래금액을 계산한 후, 평큔 거래금액이 높은 상위 5개 지역을 출력하는 코드 작성
average_price = df.groupby('시군구')['거래금액(만원)'].mean()
top_5 = average_price.sort_values(ascending=False).head(5)
top_5



시군구
서울특별시 종로구 신문로2가   322,714
서울특별시 중구 장충동1가    251,746
서울특별시 강남구 압구정동    207,247
서울특별시 용산구 용산동5가   188,139
서울특별시 종로구 홍파동     172,629
Name: 거래금액(만원), dtype: float64

In [26]:
# 시군구 열에서 '서울특별시 강남구 신사동'의 거래 데이터만 필터링한 후, 해당 데이터에서 평균 거래금액과 최대 거래금액을 계산하는 코드 작성
df2 = df[df['시군구'] == '서울특별시 강남구 신사동']
max_price = df2['거래금액(만원)'].max()
mean_price = df2['거래금액(만원)'].mean()
print(max_price)
print(mean_price)

605000
85308.32541322314
